api/utils.py

```diff
@@ -617,3 +510,524 @@ def validate_query_string(
             if not key.startswith("tag.") and not key.startswith("tag_exists."):
                 return False
     return True
+
+
+@tracer.capture_method(capture_response=False)
+def serialise_neptune_obj(obj: dict, key_prefix: str = "") -> dict:
+    """Return a new dict with properties of type dict/list serialised into string"""
+    return {
+        f"{key_prefix}{constants.SERIALISE_PREFIX if isinstance(v, (list, dict)) else ''}{k}": (
+            json.dumps(v) if isinstance(v, (list, dict)) else v
+        )
+        for k, v in obj.items()
+    }
+
+
+@tracer.capture_method(capture_response=False)
+def deserialise_neptune_obj(obj: dict) -> dict:
+    """Modifies Neptune JSON response back to hierachical JSON structure required by API"""
+    _obj = deepcopy(obj)  # Work on a copy to avoid mutation
+    for prop_name, prop_value in list(_obj.items()):
+        if prop_name.startswith(constants.SERIALISE_PREFIX):
+            actual_name = prop_name[len(constants.SERIALISE_PREFIX) :]
+            _obj[actual_name] = json.loads(prop_value)
+            del _obj[prop_name]
+        elif isinstance(prop_value, dict):
+            _obj[prop_name] = deserialise_neptune_obj(prop_value)
+    return _obj
```

Comment:
```suggestion
    deserialised = {}
    for prop_name, prop_value in obj.items():
        if prop_name.startswith(constants.SERIALISE_PREFIX):
            actual_name = prop_name[len(constants.SERIALISE_PREFIX):]
            deserialised[actual_name] = json.loads(prop_value)
        elif isinstance(prop_value, dict):
            deserialised[prop_name] = deserialise_neptune_obj(prop_value)
        else:
            deserialised[prop_name] = prop_value
    return deserialised
```

I think this should be more efficient than using deepcopy. Particularly in recursion cases.